pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -59,6 +59,7 @@ func (s *Service) ReconcileLaunchTemplate(
 	bootstrapData, err := scope.GetRawBootstrapData()
 	if err != nil {
 		record.Eventf(scope.GetMachinePool(), corev1.EventTypeWarning, "FailedGetBootstrapData", err.Error())
+		return err
```

Comment:
Its probably worth logging as well here.

Reply:
In the whole function, errors are only returned, not logged. The callers then log it like this:

```
if err := ec2Svc.ReconcileLaunchTemplate(machinePoolScope, canUpdateLaunchTemplate, runPostLaunchTemplateUpdateOperation); err != nil {
	r.Recorder.Eventf(machinePoolScope.AWSMachinePool, corev1.EventTypeWarning, "FailedLaunchTemplateReconcile", "Failed to reconcile launch template: %v", err)
	machinePoolScope.Error(err, "failed to reconcile launch template")
	return err
}
```

I prefer keeping it this way since we don't get two log lines about the same error.

Reply:
Sounds good to me.